bluebottle/accounting/views.py

```diff
@@ -0,0 +1,241 @@
+
+    def get_context_data(self, **kwargs):
+        form = kwargs.get('form')
+
+        if form and form.is_valid():
+            start = form.get_start()
+            stop = form.get_stop()
+            tenant = form.cleaned_data.get('tenant', None)
+
+            if tenant:
+                connection.set_tenant(tenant)
+                header = ' - {}'.format(tenant.name)
+                statistics = get_accounting_statistics(start, stop)
+            else:
+                header = ' - All tenants'
+                statistics = mydict()
+
+                for tenant in get_tenant_model().objects.all():
+                    connection.set_tenant(tenant)
+                    statistics += get_accounting_statistics(start, stop)
+        else:
+            header = ''
+            statistics = {}
+
+        context = super(MultiTenantAccountingOverviewView, self).get_context_data(**kwargs)
+
+        context.update({
+             'app_label': 'accounting',
+             'title': _('Accountancy Overview') + header,
+             'statistics': statistics,
+        })
+        return context
+
+
+class MultiTenantAccountingDashboardView(InitialDatesMixin, OnepercentOnlyPermissionMixin,FormView):
+    """
+    Same as AccountingDashboardView, but for multiple tenants.
+    Could be subclassed, but the distinction should be very clear,
+    this view can contain data from all tenants, the other view can NOT.
+    """
+    template_name = 'admin/accounting/dashboard.html'
+    #template_name = 'admin/accounting/multiadmin/multi_tenant_dashboard.html'
+    form_class = PeriodTenantForm
+
+    def form_valid(self, form):
+        return self.render_to_response(self.get_context_data(form=form))
+
+    def get_context_data(self, **kwargs):
+        """
+        Get the statistics data for the selected tenant, or for all tenants.
+        In the last case, the connection will be switched for each tenant,
+        and the data for each tenant will be added to the total.
+        """
+        form = kwargs.get('form')
+
+        all_tenants = None
+
+        if form and form.is_valid():
+            self.selected_start = form.get_start()
+            self.selected_stop = form.get_stop()
+            tenant = form.cleaned_data.get('tenant', None)
+
+            if tenant:
+                connection.set_tenant(tenant)
+                statistics = get_accounting_statistics(self.selected_start, self.selected_stop)
+                totals = get_dashboard_values(self.selected_start, self.selected_stop)
+
+                header = ' - {}'.format(tenant.name)
+            else:
+                header = ' - All tenants'
+                statistics = mydict()
+                totals = mydict()
+                all_tenants = {}  # contains data for each tenant when all are selected
+
+                for tenant in get_tenant_model().objects.all():
+                    connection.set_tenant(tenant)
```

Comment:
This should probably reset the tenant. I think we have a context manager for that.